pkg/archive/paths_test.go

```diff
@@ -68,3 +70,39 @@ func TesWalkZipFiles(t *testing.T) {
 		assert.NotEmpty(t, paths)
 	})
 }
+
+func TestCheckArchiveType(t *testing.T) {
+	tests := []struct {
+		name     string
+		filename string
+		want     FormatType
+		ok       bool
+	}{
+		{name: "filename.zip", filename: "filename.zip", want: ZipArchive, ok: true},
+		{name: "fat_jar.jar", filename: "fat_jar.jar", want: ZipArchive, ok: true},
+		{name: "many.dots.with.jar", filename: "many.dots.with.jar", want: ZipArchive, ok: true},
+		{name: "par_file.par", filename: "par_file.par", want: ZipArchive, ok: true},
+		{name: "generic.tar", filename: "generic.tar", want: TarArchive, ok: true},
+		{name: "many.dots.tar", filename: "many.dots.tar", want: TarArchive, ok: true},
+		{name: "compressed.tar.gz", filename: "compressed.tar.gz", want: TarGzArchive, ok: true},
+		{name: "many.dots.tar.gz", filename: "many.dots.tar.gz", want: TarGzArchive, ok: true},
+		{name: "compressed.tgz", filename: "compressed.tgz", want: TarGzArchive, ok: true},
+		{name: "bz2compressed.tar.bz2", filename: "bz2compressed.tar.bz2", want: TarBz2Archive, ok: true},
+		{name: "many.dots.tar.bz2", filename: "many.dots.tar.bz2", want: TarBz2Archive, ok: true},
+		{name: "bz2compressed.tbz2", filename: "bz2compressed.tbz2", want: TarBz2Archive, ok: true},
+		{name: "unsupported.jpg", filename: "unsupported.jpg", want: UnsupportedArchive, ok: false},
+		{name: "file.with.many.extensions", filename: "file.with.many.extensions", want: UnsupportedArchive, ok: false},
+		{name: "no-extension", filename: "no-extension", want: UnsupportedArchive, ok: false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, ok := CheckArchiveType(tt.filename)
+			if got != tt.want {
+				t.Errorf("CheckArchiveType() got = %v, want %v", got, tt.want)
+			}
+			if ok != tt.ok {
+				t.Errorf("CheckArchiveType() ok = %v, want %v", ok, tt.ok)
+			}
```

Comment:
Let's use the `assert` or `require` package to be consistent across the repo.

Reply:
Will switch in latest revision.